arroyo/processing/strategies/streaming/transform.py

```diff
@@ -268,8 +306,7 @@ def parallel_transform_worker_apply(
                 break
         else:
             i += 1
-
-    return (i, output_batch)
+    return ParallelTransformResult(i, output_batch, bad_messages)
 
 
 class ParallelTransformStep(ProcessingStep[TPayload]):
```

Comment:
Please have a unit test for this class with the changes you added. There should be already a unit test that tests this class in isolation.

Reply:
Done, I commented what I think is going on with the parallel processes asserts